proposals/mainnet_migration.md

```diff
@@ -0,0 +1,528 @@
+- Verify that the transactions in the payload yield the [transaction root](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0032.md#specification) provided in the block header.
+
+If any of the steps fails, the block is invalid.
+
+#### Snapshot Block
+
+The snapshot block `b` is constructed in accordance with the definition of _[genesis block](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0060.md#genesis-block)_ in LIP 0060\. The details for the `header` and `assets` property are specified in the following subsections.
+
+##### Header
+
+Let `a` be the block at height `HEIGHT_SNAPSHOT`. Then the following points define how `b.header` is constructed:
+
+- `b.header.version = SNAPSHOT_BLOCK_VERSION`
+- `b.header.timestamp = a.header.timestamp + SNAPSHOT_TIME_GAP`
+- `b.header.height = HEIGHT_SNAPSHOT+1`
+- `b.header.previousBlockID = blockID(a)`
+- all other block header properties must be as specified in [LIP 0060](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0060.md#validation-2)
+
+##### Assets
+
+From the [registered modules](#modules), the following ones add an entry to `b.assets`:
+
+- Legacy module
+- Token module
+- Auth module
+- DPoS module
+
+How these modules construct their entry for the `assets` property is defined in the next subsections. The verification of their entries is defined in the _Genesis Block Processing_ sections of the respective module LIPs. Note that once all modules add their entries to `b.assets`, this array [must be sorted by increasing value of moduleID](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0055.md#validation-1).
+
+In the following, let `accountsState` be the key-value store of the accounts state of `STATE_SNAPSHOT` computed as described [above](#process). That means, for a byte array `addr` representing an address, `accountsState[addr]` is the corresponding account object following the account schema defined in [LIP 0030](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0030.md). Moreover, let `accounts` be the array that contains all values of `accountsState` for which the key is a 20-byte address sorted in lexicographical order of their `address` property. Correspondingly, let `legacyAccounts` be the array that contains all values of `accountsState` for which the key is an 8-byte address sorted in lexicographical order of their `address` property.
+
+###### Assets Entry for Legacy Module
+
+The `assets` entry for the legacy module is added by the logic defined in the function `addLegacyModuleEntry` in the following pseudo code:
+
+```python
+addLegacyModuleEntry():
+    legacyObj = {}
+    legacyObj.accounts = []
+    for every account in legacyAccounts:
+        userObj = {}
+        userObj.address = account.address
+        userObj.balance = account.token.balance
+        legacyObj.accounts.append(userObj)
+    data = serialization of legacyObj using genesisLegacyStoreSchema
```

Comment:
The reference URL to `genesisLegacyStoreSchema` is missing. Please consider adding them as a note below the code block.

Reply:
Added a reference to the schema.